### PR TITLE
Consolidate release notes for 3.6

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Also add a couple of missing documentation entries for the new pages.

## PR Checklist

**Tests and Styling**
- [n/a] Has pytest style unit tests (and `pytest` passes).
- [n/a] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [n/a] New features are documented, with examples if plot related.
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+been added to the figure. Invoking the incompatible layout engine used to warn,
+but now raises with a ``RuntimeError``.
+
+``CallbackRegistry`` raises on unknown signals
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When Matplotlib instantiates a `.CallbackRegistry`, it now limits callbacks to
+the signals that the registry knows about. In practice, this means that calling
+`~.FigureCanvasBase.mpl_connect` with an invalid signal name now raises a
+`ValueError`.
```
Comment (anntzer) [inline doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst:252]: Markup for builtin exception classes is inconsistent between e.g. here and the next entry.

File: doc/api/prev_api_changes/api_changes_3.6.0/deprecations.rst [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+*offset_transform* (consistently with the property name).
+
+Calling ``MarkerStyle()`` with no arguments or ``MarkerStyle(None)`` is
+deprecated; use ``MarkerStyle("")`` to construct an empty marker style.
+
+``Axes.get_window_extent`` / ``Figure.get_window_extent`` accept only
+*renderer*. This aligns the API with the general `.Artist.get_window_extent`
+API. All other parameters were ignored anyway.
+
+This *cleared* parameter of ``get_renderer``, which only existed for AGG-based
```
Comment (anntzer) [inline doc/api/prev_api_changes/api_changes_3.6.0/deprecations.rst:137]: This -> The

File: doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst (hunk 1 above)
Comment (QuLogic) [reply doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst:252]: Fixed.

File: doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+`~matplotlib.axes.Axes.contourf` have a new keyword argument *algorithm* to
+control which algorithm is used to calculate the contours. There is a choice of
+four algorithms to use, and the default is to use ``algorithm='mpl2014'`` which
+is the same algorithm that Matplotlib has been using since 2014.
+
+Other possible values of the *algorithm* keyword argument are ``'mpl2005'``,
+``'serial'`` and ``'threaded'``; see the `ContourPy documentation
+<https://contourpy.readthedocs.io>`_ for further details.
+
+Default date limits changed to 1970-01-01 to 1970-01-02
```
Comment (timhoffm) [inline doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst:35]: The double "to" is slightly confusing. Maybe

```suggestion
Default date limits changed to 1970-01-01 - 1970-01-02
```

File: doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+canvas is now the sole child widget; the toolbar is now a regular toolbar added
+using ``SetToolBar``.
+
+Incompatible layout engines raise
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``tight_layout`` and ``constrained_layout`` are incompatible if a colorbar has
+been added to the figure. Invoking the incompatible layout engine used to warn,
+but now raises with a `RuntimeError`.
+
```
Comment (jklymak) [inline doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst:245]: Sorry, this was my wording, but if we can clarify:

```suggestion
You cannot switch between ``tight_layout`` and ``constrained_layout`` if a colorbar has
already been added to a figure. Invoking the incompatible layout engine used to warn,
but now raises with a `RuntimeError`.
```

File: doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+between the contour levels, as each polygon represents the location of values
+that lie between two levels.
+
+Axes title now avoids y-axis offset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Previously, Axes titles could overlap the y-axis offset text, which is often in
+the upper left corner of the axes. Now titles are moved above the offset text
+if overlapping, and automatic positioning is in effect (i.e. if *y* in
+`.Axes.set_title` is *None* and :rc:`axes.titley` is also *None*).
```
Comment (jklymak) [inline doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst:96]:
```suggestion
Axes title now avoids y-axis offset
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Previously, Axes titles could overlap the y-axis offset text, which is often in
the upper left corner of the axes. Now titles are moved above the offset text
if overlapping when automatic title positioning is in effect (i.e. if *y* in
`.Axes.set_title` is *None* and :rc:`axes.titley` is also *None*).
```

File: doc/users/prev_whats_new/whats_new_3.6.0.rst [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+passing the ratios in *gridspec_kws* arguments.
+
+New ``layout_engine`` module
+----------------------------
+
+Matplotlib ships with ``tight_layout`` and ``constrained_layout`` layout
+engines.  A new `.layout_engine` module is provided to allow downstream
+libraries to write their own layout engines and `~.figure.Figure` objects can
+now take a `.LayoutEngine` subclass as an argument to the *layout* parameter.
+
```
Comment (jklymak) [inline doc/users/prev_whats_new/whats_new_3.6.0.rst:32]: Ack!  Compressed Layout https://github.com/matplotlib/matplotlib/pull/22289 went in with no what's-new!  Would you like me to write one here, or write a separate one?

Comment (QuLogic) [reply doc/users/prev_whats_new/whats_new_3.6.0.rst:32]: Leave a note here and I will include it as I go through the rest of the PRs.

Comment (jklymak) [reply doc/users/prev_whats_new/whats_new_3.6.0.rst:32]: I'll have one tonight. Sorry for the extra bother.

Comment (jklymak) [inline doc/users/prev_whats_new/whats_new_3.6.0.rst:32]:
```suggestion
```suggestion

Compressed layout for fixed-aspect ratio axes
----------------------------------------------

Simple arrangements of axes with fixed aspect ratios can now be packed together with 
``fig, axs = plt.subplots(2, 3, layout='compressed')``.  With ``layout='tight'` or `'constrained'`, axes
with a fixed aspect ratio can leave large gaps between the axes.  Using ``layout='compressed'`` reduces 
the space between the axes, and adds the extra space to the outer margins.  See `compressed_layout_`.
```

You may need to check the rst link.  The anchor is `_compressed_layout` in `tutorials/intermediate/constrainedlayout_guide.py`.

**Comment by QuLogic** [comment]
Some additional PRs:
- #22607
- #22556
- #22452 
- #22387
- #22289 (above)
- #22180
- #22163
- #22013 maybe?
- #21975
- #21790
- #21784
- #21542
- #21512
- #21274
- #22936
- #22928
- #23266 
- #23644 maybe?
- #23475
- #23712

File: doc/users/prev_whats_new/whats_new_3.6.0.rst [hunk 7]
```diff
@@ -0,0 +1,720 @@
...
+it can be used as it is used with `.Line2D` objects.
+
+Rectangle patch rotation point
+------------------------------
+
+The rotation point of the `~matplotlib.patches.Rectangle` can now be set to
+'xy', 'center' or a 2-tuple of numbers.
+
+Axes.inset_axes flexibility
+---------------------------
```
Comment (rcomer) [inline doc/users/prev_whats_new/whats_new_3.6.0.rst:280]: I don’t know how much rearranging you generally do at this stage, but I wonder if this should go under “Figure and Axes creation / management”

**Comment by tacaswell** [comment]
- remove the errorbar markerfacecoloralt from api changes (it is in whats new)
- merge the contour/contourf new kwarg into the whats new entry (rather than an API behavior change)

**Review by tacaswell — APPROVED** [review]

**Pull request merged**